src/exception/call_stack.cr

```diff
@@ -17,18 +17,90 @@ def caller : Array(String)
   Exception::CallStack.new.printable_backtrace
 end
 
-# :nodoc:
+# Returns the current execution stack as an array of `Exception::CallStack::Frame`.
+def caller_frames : Array(Exception::CallStack::Frame)
+  Exception::CallStack.new.frames
+end
+
 struct Exception::CallStack
+  # Represents a specific frame within an `Exception#backtrace`;
+  # exposing the location, function name, and memory address (if available).
+  struct Frame
+    # Returns the file that at which the exception occurred.
+    #
+    # Assuming a frame like `/home/crystal/test.cr:24:5 in 'test_method' at 0x55df02499e76`,
+    # the file would be `/home/crystal/test.cr`.
+    getter file : String
+
+    # Returns the function name in which the exception occurred.
+    #
+    # Assuming a frame like `/home/crystal/test.cr:24:5 in 'test_method' at 0x55df02499e76`,
+    # the function would be `test_method`.
+    getter function : String
+
+    # Returns the line number in which the exception occurred.
+    #
+    # Assuming a frame like `/home/crystal/test.cr:24:5 in 'test_method' at 0x55df02499e76`,
+    # the line number would be `24`.
+    getter line_number : Int32
+
+    # Returns the column number in which the exception occurred.
+    #
+    # Assuming a frame like `/home/crystal/test.cr:24:5 in 'test_method' at 0x55df02499e76`,
+    # the column number would be `5`.
+    getter column_number : Int32
+
+    # Returns the memory address
```

Comment:
I'm assuming this is the memory address, but I'm not actually what the address points to specifically.